src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs

```diff
@@ -107,12 +110,56 @@ public async Task<IEnumerable<string>> ExtractPackageAsync(string packagePath, s
             NuGetPackagePathResolver packagePathResolver = new NuGetPackagePathResolver(targetFolder);
             CancellationToken cancellationToken = CancellationToken.None;
 
-            return await PackageExtractor.ExtractPackageAsync(
+            var allFilesInPackage = await PackageExtractor.ExtractPackageAsync(
                 targetFolder,
                 packageStream,
                 packagePathResolver,
                 packageExtractionContext,
                 cancellationToken);
+
+            if (!OperatingSystem.IsWindows())
+            {
+                foreach (FilePath filePath in FindAllFilesNeedExecutablePermission(allFilesInPackage, targetFolder))
+                {
+                    _filePermissionSetter.Set755Permission(filePath.Value);
+                }
+            }
+
+            return allFilesInPackage;
+        }
+
+        internal IEnumerable<FilePath> FindAllFilesNeedExecutablePermission(IEnumerable<string> files,
+            string targetPath)
+        {
+            if (!PackageIsInAllowList(files))
```

Comment:
Is this a permanent solution? It seems like the allow list will get quickly out of hand when there are more workloads, are we planning on implementing a way for workloads to provide their own allow list?